server/src/game/role/marksman.rs

```diff
@@ -0,0 +1,208 @@
+    }
+}
+
+pub(super) const FACTION: Faction = Faction::Town;
+pub(super) const MAXIMUM_COUNT: Option<u8> = None;
+
+impl RoleStateImpl for Marksman {
+    fn defense(&self, _game: &Game, _actor_ref: PlayerReference) -> u8 {0}
+    
+
+    fn do_night_action(mut self, game: &mut Game, actor_ref: PlayerReference, priority: Priority) {
+        
+    
+        match priority{
+            Priority::Kill => {
+                let visiting_players: Vec<_> = actor_ref
+                    .night_visits(game)
+                    .into_iter()
+                    .flat_map(|p|p.target.all_visitors(game))
+                    .collect();
+
+                for mark in self.state.marks().into_iter() {
+                    
+                    if !visiting_players.contains(&mark) {continue};
+                    
+                    actor_ref.push_night_message(game, ChatMessageVariant::MarksmanAttackedVisitor);
+                    let killed = mark.try_night_kill(actor_ref, game, GraveKiller::Role(Role::Marksman), 1, false);
+
+                    if killed && mark.role(game).faction() == Faction::Town {
+                        self.state = MarksmanState::ShotTownie;
+                    }
+                }
+                
+                actor_ref.set_role_state(game, RoleState::Marksman(self));
+            },
+            _ => {}
+        }
+
+    }
+    fn do_day_action(mut self, game: &mut Game, actor_ref: PlayerReference, target_ref: PlayerReference) {
+        self.state.toggle_mark(target_ref);
+        if self.state.marks().len() == 0 {
+            actor_ref.set_chosen_targets(game, vec![]);
+        }
+        actor_ref.add_private_chat_message(game, ChatMessageVariant::MarksmanChosenMarks { marks: PlayerReference::ref_vec_to_index(self.state.marks().deref()) });
+        actor_ref.set_role_state(game, RoleState::Marksman(self))
+    }
+    fn can_night_target(self, game: &Game, actor_ref: PlayerReference, target_ref: PlayerReference) -> bool {
+        let chosen_targets = actor_ref.chosen_targets(game);
+        
+        !self.state.no_marks() &&
+        actor_ref != target_ref &&
+        !actor_ref.night_jailed(game) &&
+        actor_ref.alive(game) &&
+        target_ref.alive(game) && 
+        ((
+            chosen_targets.is_empty()
+        ) || (
+            chosen_targets.len() == 1 &&
+            Some(target_ref) != chosen_targets.first().copied()
+        ))
+    }
+    fn can_day_target(self, game: &Game, actor_ref: PlayerReference, target_ref: PlayerReference) -> bool {
+        game.current_phase().is_night() &&
+        target_ref.alive(game) &&
+        if let MarksmanState::Marks {..} = self.state {true}else{false} &&
+        actor_ref != target_ref
+    }
+    fn convert_targets_to_visits(self, _game: &Game, _actor_ref: PlayerReference, target_refs: Vec<PlayerReference>) -> Vec<Visit> {
+        if target_refs.len() == 2 {
+            vec![
+                Visit{ target: target_refs[0], attack: false }, 
```

Comment:
We should talk about attacking visits versus indirect visits.. this never really sat right with me and I also don't quite get it.
Just a reminder, no action required here